healthchain/models/data/concept.py

```diff
@@ -21,6 +21,25 @@ class Quantity(DataType):
     value: Optional[Union[str, float]] = None
     unit: Optional[str] = None
 
+    @field_validator("value")
+    @classmethod
+    def validate_value(cls, value: Union[str, float]):
+        if value is None:
```

Comment:
I ran the testing suite and got some errors from this line. The logic here should be slightly different. If None is passed in then we want to just return None back, however if a type other than str or float is passed then we can raise the warning. I've added a suggestion below (written on my phone so will need some edits).
```suggestion
        if value is None:
           return None
        elif type(value) is not str or float:
           raise TypeError
```

Reply:
Okay got it ! Will make the change.